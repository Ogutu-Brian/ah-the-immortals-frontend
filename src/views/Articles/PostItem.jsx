import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import Moment from 'react-moment';
import '../../styles/App.css';


class PostItem extends Component {


  render() {
    const { post } = this.props;
    const {created_at} = post;
    return (
      <div className="quotes">
        <div className="box box1">

          <div className="">
            <div className="title-header">
              {post.title}
              <div className="timer">
                <div className="bookmark">
                  <i className="far fa-bookmark" />
                </div>
                <span className="readingTime">
                  {post.readtime}
                </span>
              </div>
            </div>
            <div className="author">
              {post.author.username}
            </div>
            <div className="description">
              {post.description}
              <Link to={`/post/${post.slug}`}>
                .... Read More
              </Link>
            </div>
            <div className="date-read-time">
              <time>
                <Moment format="DD-MMM-YYYY">
                  {created_at}
                </Moment>
              </time>
            </div>
          </div>
        </div>
      </div>
      
     
    );
  }
}



PostItem.propTypes = {

  post: PropTypes.object.isRequired

};

const mapStateToProps = state => ({

});

export default connect(mapStateToProps)(
  PostItem
);
