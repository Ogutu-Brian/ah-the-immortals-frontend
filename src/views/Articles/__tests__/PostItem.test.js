import {mount, shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import MutationObserver from 'mutation-observer';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import React from 'react';
import renderHTML from 'react-render-html';
import PostItem from '../PostItem';

global.MutationObserver = MutationObserver;
document.getSelection = jest.fn();

describe('pass', () => {
  let testStore = configureMockStore([thunk]);
  let store = testStore({post:{}});
  let props = {
    post: jest.fn(),
  };
  const wrapper = shallow(
    <Provider store={store}>
      <PostItem post={props} />
    </Provider>
  );
  it('test 1', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
