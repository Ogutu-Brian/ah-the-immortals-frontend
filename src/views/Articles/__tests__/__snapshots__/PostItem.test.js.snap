// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pass test 1 1`] = `ShallowWrapper {}`;
